### PR TITLE
Update controller runtime and use networking.k8s.io

### PR DESCRIPTION
**Pull request by AMecea**
- use go.mod
 - migrate to kubebuilder v2
 - use Ingress from networking.k8s.io/v1beta1
 - cleanup old cron job used for wp cron

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -26,7 +26,8 @@ steps:
 - name: verify generated code
   image: quay.io/presslabs/bfc:build
   commands:
-  - make -j4 generate
+  - make generate
```
[the remark this answers is not in the document]
Comment (AMecea) [reply .drone.yml:29]: done

File: pkg/controller/wordpress/internal/sync/code_pvc.go [hunk 2]
```diff
@@ -41,8 +41,8 @@ func NewCodePVCSyncer(wp *wordpress.Wordpress, c client.Client, scheme *runtime.
 			Namespace: wp.Namespace,
 		},
 	}
-	return syncer.NewObjectSyncer("CodePVC", wp.Unwrap(), obj, c, scheme, func(existing runtime.Object) error {
-		out := existing.(*corev1.PersistentVolumeClaim)
+	return syncer.NewObjectSyncer("CodePVC", wp.Unwrap(), obj, c, scheme, func() error {
+		out := obj
```
Comment (calind) [inline pkg/controller/wordpress/internal/sync/code_pvc.go:45]: Just use `obj`.

File: pkg/controller/wordpress/internal/sync/deploy.go [hunk 3]
```diff
@@ -46,8 +46,8 @@ func NewDeploymentSyncer(wp *wordpress.Wordpress, secret *corev1.Secret, c clien
 		},
 	}
 
-	return syncer.NewObjectSyncer("Deployment", wp.Unwrap(), obj, c, scheme, func(existing runtime.Object) error {
-		out := existing.(*appsv1.Deployment)
+	return syncer.NewObjectSyncer("Deployment", wp.Unwrap(), obj, c, scheme, func() error {
+		out := obj
```
Comment (calind) [inline pkg/controller/wordpress/internal/sync/deploy.go:50]: Just use `obj`.

File: pkg/controller/wordpress/internal/sync/ingress.go [hunk 4]
```diff
@@ -72,20 +72,20 @@ func upsertPath(rules []extv1beta1.IngressRule, domain, path string, bk extv1bet
...
-	bk := extv1beta1.IngressBackend{
+	bk := netv1beta1.IngressBackend{
 		ServiceName: wp.ComponentName(wordpress.WordpressService),
 		ServicePort: intstr.FromString("http"),
 	}
 
-	return syncer.NewObjectSyncer("Ingress", wp.Unwrap(), obj, c, scheme, func(existing runtime.Object) error {
-		out := existing.(*extv1beta1.Ingress)
+	return syncer.NewObjectSyncer("Ingress", wp.Unwrap(), obj, c, scheme, func() error {
+		out := obj
```
Comment (calind) [inline pkg/controller/wordpress/internal/sync/ingress.go:88]: Just use `obj`.

File: pkg/controller/wordpress/internal/sync/media_pvc.go [hunk 5]
```diff
@@ -41,8 +41,8 @@ func NewMediaPVCSyncer(wp *wordpress.Wordpress, c client.Client, scheme *runtime
 			Namespace: wp.Namespace,
 		},
 	}
-	return syncer.NewObjectSyncer("MediaPVC", wp.Unwrap(), obj, c, scheme, func(existing runtime.Object) error {
-		out := existing.(*corev1.PersistentVolumeClaim)
+	return syncer.NewObjectSyncer("MediaPVC", wp.Unwrap(), obj, c, scheme, func() error {
+		out := obj
```
Comment (calind) [inline pkg/controller/wordpress/internal/sync/media_pvc.go:45]: Just use `obj`.

File: pkg/controller/wordpress/internal/sync/secret.go [hunk 6]
```diff
@@ -53,8 +53,8 @@ func NewSecretSyncer(wp *wordpress.Wordpress, c client.Client, scheme *runtime.S
 		},
 	}
 
-	return syncer.NewObjectSyncer("Secret", wp.Unwrap(), obj, c, scheme, func(existing runtime.Object) error {
-		out := existing.(*corev1.Secret)
+	return syncer.NewObjectSyncer("Secret", wp.Unwrap(), obj, c, scheme, func() error {
+		out := obj
```
Comment (calind) [inline pkg/controller/wordpress/internal/sync/secret.go:57]: Just use `obj`.

File: pkg/controller/wordpress/internal/sync/service.go [hunk 7]
```diff
@@ -42,8 +42,8 @@ func NewServiceSyncer(wp *wordpress.Wordpress, c client.Client, scheme *runtime.
 		},
 	}
 
-	return syncer.NewObjectSyncer("Service", wp.Unwrap(), obj, c, scheme, func(existing runtime.Object) error {
-		out := existing.(*corev1.Service)
+	return syncer.NewObjectSyncer("Service", wp.Unwrap(), obj, c, scheme, func() error {
+		out := obj
```
Comment (calind) [inline pkg/controller/wordpress/internal/sync/service.go:46]: Just use `obj`.

File: pkg/controller/wordpress/wordpress_controller.go [hunk 8]
```diff
@@ -185,3 +191,21 @@ func (r *ReconcileWordpress) maybeMigrate(wp *wordpressv1alpha1.Wordpress) (*wor
...
+		Name:      wp.ComponentName(wordpress.WordpressCron),
+		Namespace: wp.Namespace,
+	}
+
+	cronJob := &batchv1beta1.CronJob{}
+	if err := r.Get(context.TODO(), cronKey, cronJob); err != nil {
+		return ignoreNotFound(err)
+	}
+
+	if err := r.Delete(context.TODO(), cronJob); err != nil {
```
Comment (calind) [inline pkg/controller/wordpress/wordpress_controller.go:206]: Check if the CronJob is owned by the WordPress object.

**Pull request merged**